code/lib/cli/src/upgrade.ts

```diff
@@ -43,7 +44,14 @@ export const getStorybookVersion = (line: string) => {
 const getInstalledStorybookVersion = async (packageManager: JsPackageManager) => {
   const installations = await packageManager.findInstallations(['storybook', '@storybook/cli']);
   if (!installations) {
-    return;
+    /**
+     * In older versions of Storybook, the user may not have a direct dependency on the CLI package.
+     * In this case, we need to use the legacy method to determine the version.
+     * @see legacy_getStorybookVersion
+     * @deprecated
+     * @remove in SB9 (or when we drop support for upgrading from SB6 & SB7)
+     */
+    return legacy_getStorybookVersion();
```

Comment:
So where before we returned `undefined` which would cause an "no version detected"-error... we now have this fallback that should operate exactly how it was in SB6.